_inc/lib/core-api/class.jetpack-core-api-site-endpoints.php

```diff
@@ -92,22 +92,15 @@ public static function get_related_posts( $api_request ) {
 		// Decode the results
 		$results = json_decode( $response['body'], true );
 
-		// Bail if there were no results returned
-		if ( ! isset( $results['hits'] ) || ! is_array( $results['hits'] ) ) {
-			return new WP_Error(
-				'no_related_posts_found',
-				esc_html__( 'No related posts found for the specified post.', 'jetpack' ),
-				array( 'status' => 404 )
-			);
-		}
-
-		$response_body = json_decode( wp_remote_retrieve_body( $response ), true );
-		$related_posts_ids = array_map( array( 'Jetpack_Core_API_Site_Endpoint', 'get_related_post_id' ), $response_body['hits'] );
-
 		$related_posts = array();
-		$related_posts_instance = Jetpack_RelatedPosts::init();
-		foreach ( $related_posts_ids as $related_post_id ) {
-			$related_posts[] = $related_posts_instance->get_related_post_data_for_post( $related_post_id, 0, 0 );
+		if ( isset( $results['hits'] ) && is_array( $results['hits'] ) ) {
+			$response_body = json_decode( wp_remote_retrieve_body( $response ), true );
```

Comment:
I know you didn't modify these lines (and just wrapped them with `if`), but this line looks a bit odd since we just had this earlier at line 93:
```php
$results = json_decode( $response['body'], true );
```

Reply:
Nah, you're bringing a perfectly valid point - we have a repetition here. Good catch 👍 

I've improved it in https://github.com/Automattic/jetpack/pull/10289/commits/120fa04740a5d7a8bdc0bf023f7ea1d3a00b0091

Feel free to have a new look.